File: Js/Smallestnumber.js
var nextGreatestLetter = function (letters, target) {
  let start = 0
  let end = letters.length - 1

  while (start <= end) {
    let mid = parseInt(start + (end - start) / 2)
    if (target < letters[mid]) {
      end = mid - 1
    } else {
      start = mid + 1
    }
  }
  console.log(letters[start % letters.length])
  return letters[start % letters.length]
}

letters = ["e", "e", "e", "e", "e", "e", "n", "n", "n", "n"]
target = "a"
nextGreatestLetter(letters, target)

File: Java/Smallestnumber.java
package Java;

public class Smallestnumber {
    public static void main(String[] args) {
        char[] letters = { 'c', 'f', 'j' };
        char target = 'd';
        char charecter = getChar(letters, target);
        System.out.println(charecter);

    }

    static char getChar(char[] letters, char target) {
        int start = 0;
        int end = letters.length - 1;

        while (start <= end) {
            int mid = start + (end - start) / 2;

            if (target < letters[mid]) {
                end = mid - 1;
            } else {
                start = mid + 1;
            }

        }
        return letters[start % letters.length];

    }
}
